team_red/gen/generation_service.py

```diff
@@ -56,15 +35,16 @@ def generate(self, parameters: Dict[str, str]) -> GenResponse:
             "\n====== Resolved prompt =====\n\n%s\n\n=============================",
             resolved,
         )
-
-        response = self.pipeline(
+        response = self.set_model()(
```

Comment:
does a model have to be created for each individual `generate` request? That seems to be overkill. Why not initialise a model once as a member variable (in `set_model -> init_model`) and then call `self._model` directly?

Reply:
previously `self.pipeline` used to be a (lazily initialised) getter property. imho that produces easier to comprehend code and is more efficient than initializing a model for each request

Reply:
I changed the way the model is being initialised in the most recent commit.